src/prisma/course.ts

```diff
@@ -17,38 +18,112 @@ async function findCourseSection(courseterm: Term) {
   return courses;
 }
 
+const getMaxHoursPerWeek = (assignments: Assignment[] | undefined): number => {
+  if (!assignments) return 0;
+  return Math.max(...assignments.map((h) => h.hoursWeek));
+};
+
+const getDays = (assignment: Assignment) => {
+  let days = utils.appendDay(assignment.monday, 'MONDAY', []);
+  days = utils.appendDay(assignment.tuesday, 'TUESDAY', days);
+  days = utils.appendDay(assignment.wednesday, 'WEDNESDAY', days);
+  days = utils.appendDay(assignment.thursday, 'THURSDAY', days);
+  days = utils.appendDay(assignment.friday, 'FRIDAY', days);
+  days = utils.appendDay(assignment.saturday, 'SATURDAY', days);
+  days = utils.appendDay(assignment.sunday, 'SUNDAY', days);
+  return days;
+};
+
+const assignmentsToMeetingTime = (assignments?: Assignment[] | undefined) => {
+  if (!assignments) return [];
+  return assignments.map((assignment) => {
+    return {
+      startTime: assignment.beginTime,
+      endTime: assignment.endTime,
+      days: getDays(assignment),
+    };
+  });
+};
+
+const getMinMaxDateTime = (
+  term: Term,
+  year: number,
+  assignments: Assignment[] | undefined
+) => {
+  // example: Apr 05, 2019
+  if (!assignments) return null;
+  // assuming we get proper date objects
+
+  const dates = assignments.map((assignment) => {
+    // start
+    if (assignment.startDate !== '' && assignment.endDate !== '') {
+      return {
+        start: getDateTime(assignment.beginTime, assignment.startDate),
+        end: getDateTime(assignment.endTime, assignment.endDate),
+      };
+    }
+
+    const startTime = getTime(assignment.beginTime);
+    const endTime = getTime(assignment.endTime);
+
+    if (term === Term.FALL) {
+      startTime.setFullYear(year, 10, 1);
+      endTime.setFullYear(year, 11, 1);
+    } else if (term === Term.SPRING) {
+      startTime.setFullYear(year, 0, 1);
+      endTime.setFullYear(year, 3, 1);
+    } else if (term === Term.SUMMER) {
+      startTime.setFullYear(year, 4, 1);
+      endTime.setFullYear(year, 9, 1);
```

Comment:
endTime for Summer should be 7

Reply:
But yeah you're right about it being off. Fixing